### ISSUE TITLE
Add `when()` method on `QueryBuilder` classes

### ISSUE THREAD

**Issue by 2Seg** [babenkoivan/elastic-scout-driver-plus#120]
Hello! 👋

I'm implementing this package since **babenkoivan/scout-elasticsearch-driver** is archived. 
Don't you think it would be interesting to also implement a `when()` method on `AbstractParameterizedQueryBuilder` so that we could call it on child classes such as ```BoolQueryBuilder``` for instance?

All these child classes are final and not macroable so I can't see a way to implement it on my own.

Here is an exemple : 
```php
Query::bool()
    ->when(true, function ($query) {
        return $query->must(...)
    })
    ->when(false, function ($query) {
        return $query->filter(...)
    })
```

Thanks for the great work!
Cheers, 
Eliott

**Comment by babenkoivan** [comment]
Hi @2Seg, I've just released [v4.0.0](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v4.0.0) that supports `when` and `unless` methods for query builders 🎉

**Comment by 2Seg** [comment]
Thanks @babenkoivan!! 🤩
